src/olympia/devhub/helpers.py

```diff
@@ -115,7 +115,17 @@ def docs_breadcrumbs(context, items=None):
 @jinja2.contextfunction
 def add_file_modal(context, title, action, action_label, modal_type='file'):
     addon = context['addon']
-    upload_url = reverse('devhub.upload_for_addon', args=[addon.slug])
+    version = context.get('version',
+                          addon.find_latest_version_including_rejected())
```

Comment:
What sucks?  Not sure what you're asking here.  Fwiw, the whole `add_file_modal` is going as part of unlisted/listed but not yet.